src/os/bluestore/StupidAllocator.cc

```diff
@@ -139,10 +141,11 @@ int StupidAllocator::allocate(
     }
   }
 
-  // search down (origin)
+  // search down (from origin, and skip searched extents by hint)
```

Comment:
Ideally, we should only search up for free extents, since previous bin of _choose_bin() only contains free extents with length < need_size. I am not sure why Sage had those two "search down" loops in the first place, maybe he used this as last resort in case code has bugs? @liewegas do you think, we should remove those two "search down" loops at all?